### PR TITLE
fix: [DEL-6029]: new command type to fetch k8s installation instructions 

### PR DESCRIPTION
**Pull request by xingchi-jin**
Background:
We recently received feedback from the field that the installation wizard for k8s manifest is not well designed. Customers struggled because it's hard for them to find correct values and replace in the yaml template. This PR provides an API option to show instructions with populated values.

In details, it includes
1. add kubernetes command to get k8s install instructions 
2. remove terraform command
3. Always who immutable delegate yaml for k8s ng 

Two other simple fixes also included
1. set failureThreshold: 3, to keep the yaml generated same as the one from helm chart and from the k8s manifest git repo
2. Remove the javaOps setting in terraform example module template file 

Test:
LOCAL: API test
image
LOCAL - k8s yaml generation ng
I set immutable flag in account to be false
image

The k8s yaml generated is still for immutable delegate
image

PR-ENV - immutable flag in db is false
1. Tested with downloading CG delegates: docker, helm, kubernetes - None of them are affected by this change. Still providing legacy delegate files
2. In ng, it is always the immutable yaml be downloaded regardless of the flag 
image

PR -ENV - terraform file 
no javaOps 
image

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44361)

### CODE REVIEW

File: 400-rest/src/main/resources/delegatetemplates/harness-delegate-ng-immutable.yaml.ftl [hunk 1]
```diff
@@ -92,7 +92,7 @@ spec:
             scheme: HTTP
           initialDelaySeconds: 10
           periodSeconds: 10
-          failureThreshold: 2
+          failureThreshold: 3
```
Comment (xingchi-jin) [inline 400-rest/src/main/resources/delegatetemplates/harness-delegate-ng-immutable.yaml.ftl:95]: This line is changed to keep the file consistent with helm value. https://github.com/harness/delegate-helm-chart/blob/main/harness-delegate-ng/values.yaml#L64

**Review by Vbhatia20 — APPROVED** [review]
LGTM
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44361#-:-NPJj8HY9SDzptKpdICm:bb6ywmx)* status: 4 file(s) to review. 0 of 1 approvals obtained. Waiting on team(s): del-team.

**Comment by xingchi-jin** [comment]
trigger ut1

**Review by AnupamIO — COMMENTED** [review]
> Reviewed 1 of 1 files at r3, 1 of 1 files at r4, 2 of 2 files at r5, 3 of 5 files at r6, 1 of 1 files at r7, 3 of 3 files at r8, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44361)* status: 0 of 1 approvals obtained. Waiting on team(s): del-team. 

___
*[`400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java` line 1470 at r9](https://reviewable.io/reviews/harness/harness-core/44361#-NPLZjKZ0WEdpnkQwfLz:-NPLZjKZ0WEdpnkQwfM-:bkmd7yq) ([raw file](https://github.com/harness/harness-core/blob/97e4c9012d6b6316633601f3cbab355b3009d751/400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java#L1470)):*
> ```Java
>         (isNgDelegate
>             && (HELM_DELEGATE.equals(templateParameters.getDelegateType())
>                 || KUBERNETES.equals(templateParameters.getDelegateType())))
> ```

- if someone wants legacy yaml in ng for k8 delegate, how will we do it? 

- what about docker based installation, we should have same type of behaviour for docker and k8. Giving legacy image in docker and immutable in k8 looks confusing.

- Why not write a migration to migrate immutable variable in accounts collection for ng accounts ? One issue I see here is, this will change CG behaviour as well. If we are sure that nextGenEnabled in accounts collections means that customer uses NG only then we can use migration.

**Review by xingchi-jin — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44361)* status: 1 file(s) to review. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java` line 1470 at r9](https://reviewable.io/reviews/harness/harness-core/44361#-NPLZjKZ0WEdpnkQwfLz:-NPOOmor02ATkgFq0qOE:b-q4noqg) ([raw file](https://github.com/harness/harness-core/blob/97e4c9012d6b6316633601f3cbab355b3009d751/400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java#L1470)):*
<details><summary><i>Previously, AnupamIO (Anupam Jaiswal) wrote…</i></summary><blockquote>

- if someone wants legacy yaml in ng for k8 delegate, how will we do it? 

- what about docker based installation, we should have same type of behaviour for docker and k8. Giving legacy image in docker and immutable in k8 looks confusing.

- Why not write a migration to migrate immutable variable in accounts collection for ng accounts ? One issue I see here is, this will change CG behaviour as well. If we are sure that nextGenEnabled in accounts collections means that customer uses NG only then we can use migration. 

</blockquote></details>

1. we don't give legacy yaml in ng
2. Will do in separate PR, we plan to have this fix hotfixed. only for k8s ng
3. why it will affect CG?  CG comes with a different code path for k8s

**Review by raghuAtWings — APPROVED** [review]
> Reviewed 1 of 1 files at r3, 1 of 1 files at r4, 2 of 2 files at r5, 3 of 5 files at r6, 1 of 1 files at r7, 3 of 3 files at r8, 1 of 1 files at r9, 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44361)* status: 1. unresolved discussions.

**Review by Vbhatia20 — COMMENTED** [review]
with this change, the image will be picked from respective ring which the account belongs to?

**Review by Vbhatia20 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44361)* status: 1 change requests. 1. unresolved discussions. 

___
*[`400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java` line 4198 at r10](https://reviewable.io/reviews/harness/harness-core/44361#-NPPEdZdAS0pkA5AV3Dp:-NPPEdZdAS0pkA5AV3Dq:bkv3gmp) ([raw file](https://github.com/harness/harness-core/blob/d209df609c5f8689405cf3c7c51f1a63964e6469/400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java#L4198)):*
> ```Java
> 
>       File yaml = File.createTempFile(HARNESS_DELEGATE, YAML);
>       String templateName = IMMUTABLE_DELEGATE_YAML;
> ```

We are getting legacy yaml with harness-delegate-ng namespace which breaks consistency as it being CG

**Review by xingchi-jin — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44361)* status: 1 change requests. 1. unresolved discussions. 

___
*[`400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java` line 4198 at r10](https://reviewable.io/reviews/harness/harness-core/44361#-NPPEdZdAS0pkA5AV3Dp:-NPPO8ak31Rp_F7aTSuv:bsxk47v) ([raw file](https://github.com/harness/harness-core/blob/d209df609c5f8689405cf3c7c51f1a63964e6469/400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java#L4198)):*
<details><summary><i>Previously, Vbhatia20 (Vishal Bhatia) wrote…</i></summary><blockquote>

We are getting legacy yaml with harness-delegate-ng namespace which breaks consistency as it being CG
</blockquote></details>

this is ng yaml template 



_Code quote:_
```Java
IMMUTABLE_DELEGATE_YAML
```

**Review by Vbhatia20 — APPROVED** [review]
> Reviewed 1 of 1 files at r3, 1 of 1 files at r4, 2 of 2 files at r5, 3 of 5 files at r6, 1 of 1 files at r7, 3 of 3 files at r8, 1 of 1 files at r9, 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44361)* status: 1. unresolved discussions.

**Review by AnupamIO — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44361#-:-NPQVbzn17Do1XFeB4D1:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**